File: src/main/java/me/leodev/trollingfreedomreborn/commands/SlipperyHands.java
package me.leodev.trollingfreedomreborn.commands;

import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;
import org.bukkit.inventory.ItemStack;

public class SlipperyHands implements Listener {


    public static void SlipperyHands(Player p) {
        int id = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, new Runnable() {
            public void run() {
                ItemStack item = p.getInventory().getItemInMainHand();
                if (p.getInventory().getItemInMainHand() != null && p.getInventory().getItemInMainHand().getType() != Material.AIR) {
                    p.getWorld().dropItemNaturally(p.getLocation(), item);
                    p.getInventory().remove(item);
                }
            }
        }, 5L, 40L);
        Core.instance.addTask(p, "slipperyhands", id);
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/random/SlipperyHands.kt
package com.leomadrassi.trollingfreedomreborn.trolls.random

import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.Listener
import org.bukkit.inventory.ItemStack

class SlipperyHands : Listener {
    companion object {
        fun SlipperyHands(p: Player) {
            val id = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, Runnable {
                val item = p.inventory.itemInMainHand
                if (p.inventory.itemInMainHand != null && p.inventory.itemInMainHand.type != Material.AIR) {
                    p.world.dropItemNaturally(p.location, item)
                    p.inventory.remove(item)
                }
            }, 5L, 40L)
            Core.instance.addTask(p, "slipperyhands", id)
        }
    }
}
